### ISSUE TITLE
`./gradlew setup` exception

### ISSUE THREAD

**Issue by TheAwesome98-Real** [MCPHackers/RetroMCP-Java#5]
```
$ ./gradlew setup

> Task :compileJava FAILED
/path/to/RetroMCP-Java/src/main/java/org/mcphackers/mcp/tasks/TaskDecompile.java:80: error: reference to newFileSystem is ambiguous
                                        MappingUtil.modifyClasses(mappingTree, FileSystems.newFileSystem(originalJar, null).getPath("/"), className -> {
                                                                                          ^
  both method newFileSystem(Path,ClassLoader) in FileSystems and method newFileSystem(Path,Map<String,?>) in FileSystems match
/path/to/RetroMCP-Java/src/main/java/org/mcphackers/mcp/tools/FileUtil.java:40: error: reference to newFileSystem is ambiguous
                try(FileSystem fs = FileSystems.newFileSystem(sourceZip, null)) {
                                               ^
  both method newFileSystem(Path,ClassLoader) in FileSystems and method newFileSystem(Path,Map<String,?>) in FileSystems match
/path/to/RetroMCP-Java/src/main/java/org/mcphackers/mcp/tools/FileUtil.java:52: error: reference to newFileSystem is ambiguous
                try(FileSystem fs = FileSystems.newFileSystem(sourceZip, null)) {
                                               ^
  both method newFileSystem(Path,ClassLoader) in FileSystems and method newFileSystem(Path,Map<String,?>) in FileSystems match
/path/to/RetroMCP-Java/src/main/java/org/mcphackers/mcp/tools/FileUtil.java:59: error: reference to newFileSystem is ambiguous
                try(FileSystem fs = FileSystems.newFileSystem(sourceZip, null)) {
                                               ^
  both method newFileSystem(Path,ClassLoader) in FileSystems and method newFileSystem(Path,Map<String,?>) in FileSystems match
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
4 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 962ms
1 actionable task: 1 executed

```

**Comment by PhoenixVX** [comment]
For Java 8+, you'll need to hardcode a cast.

**Comment by TheAwesome98-Real** [comment]
what do i cast it to?

**Comment by PhoenixVX** [comment]
ClassLoader IIRC. Don't use Java 17 for compiling with Gradle. You can use it to run for debugging, but the Gradle JVM should be Java 8.

**Comment by TheAwesome98-Real** [comment]
how do i change it? i have a jvm8 in /opt/java8/bin/java

**Comment by PhoenixVX** [comment]
Best way it to use your IDE settings, or alternatively, manually set the PATH and JAVA_HOME variables as necessary.
For Intellij, click on the Gradle vertical taskbar on the right, select the cog/settings icon at the top of the pane that opens, and set the Java version to a different version.
As for Eclipse, I am not too sure of the steps involved for that, but I do know it is possible.

**Comment by PhoenixVX** [comment]
You'll need to set JAVA_HOME and the PATH variable. The JAVA_HOME path should be above bin, lib, and the other folders. The PATH environment variable requires the bin path. For example, if I had Java at `C:/Java`, I would do
```
set JAVA_HOME=C:/Java
set PATH=C:/Java/bin;%PATH%
```

As for Linux, it would be something like this:
```
export JAVA_HOME=/usr/java
export PATH=/usr/java/bin:$PATH
```

**Comment by PhoenixVX** [comment]
Fixed in [#6df5355bd472693b4fbfb54f51efb6478f6b8d4d](https://github.com/MCPHackers/RetroMCP-Java/commit/6df5355bd472693b4fbfb54f51efb6478f6b8d4d). Closing issue, thanks for reporting!
